File: binding/src/network/libqhttp1configuration.cpp
#include <QHttp1Configuration>
#include <qhttp1configuration.h>
#include "libqhttp1configuration.h"
#include "libqhttp1configuration.hxx"

QHttp1Configuration* QHttp1Configuration_new() {
	 return new QHttp1Configuration();
}

QHttp1Configuration* QHttp1Configuration_new2(const QHttp1Configuration* other) {
	 return new QHttp1Configuration(*other);
}

void QHttp1Configuration_OperatorAssign(QHttp1Configuration* self, const QHttp1Configuration* other) {
	self->operator=(*other);
}

void QHttp1Configuration_SetNumberOfConnectionsPerHost(QHttp1Configuration* self, ptrdiff_t amount) {
	self->setNumberOfConnectionsPerHost(amount);
}

ptrdiff_t QHttp1Configuration_NumberOfConnectionsPerHost(const QHttp1Configuration* self) {
	return self->numberOfConnectionsPerHost();
}

void QHttp1Configuration_Swap(QHttp1Configuration* self, QHttp1Configuration* other) {
	self->swap(*other);
}

void QHttp1Configuration_Delete(QHttp1Configuration* self) {
    delete self;
}

File: binding/src/network/libqhttp1configuration.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qhttp1configuration.html
pub const qhttp1configuration = struct {

    /// New constructs a new QHttp1Configuration object.
    pub fn New() QtC.QHttp1Configuration {
        return qtc.QHttp1Configuration_new();
    }


    /// New2 constructs a new QHttp1Configuration object.
    pub fn New2(other: ?*anyopaque) QtC.QHttp1Configuration {
        return qtc.QHttp1Configuration_new2(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QHttp1Configuration_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn SetNumberOfConnectionsPerHost(self: ?*anyopaque, amount: i64) void {
        qtc.QHttp1Configuration_SetNumberOfConnectionsPerHost(@ptrCast(self), amount);
    }

    pub fn NumberOfConnectionsPerHost(self: ?*anyopaque, ) i64 {
        return qtc.QHttp1Configuration_NumberOfConnectionsPerHost(@ptrCast(self));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QHttp1Configuration_Swap(@ptrCast(self), @ptrCast(other));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QHttp1Configuration_Delete(@ptrCast(self));
    }
};
